### ISSUE TITLE
Text is stripped from the string literal that defines inputs for a Jest parameterised test

### ISSUE THREAD

**Issue by duncanawerbuck** [prettier/prettier#7007]
**Prettier 1.19.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuE8DOMB0cCGYAWABgDpQAEZAlgCYVkA+ZA7rgA7QDC+ElYcpFACTAADAF8GZYQHIoAVygE5YANZppEumQD025vgCeZHAFsyASTIYATpVas41APwCpwAIwTGMgEYQrMDgAZkEaWuFMhsZmljZ2Ds6uwgBMXm7SEAA2tD6ZOFAqcDBhFNoAVPpGphYBtvaOTmRl2qSEABTSbegwZGA4aHBSNACUZEEQ-pRoZD441voIZKx5BnDzODCBqtNMlDD4Uiw47FBcPHzSADRkbaMAvAB8ZMCuumTdMxDUBq5wAB4OMAwNruYaYGAQABCcC61jkcGGAG5SGIkSBLiAIKwYJRoGhkKA5tYIEwAApzBD4lA4ABuPGo6JAPmseEKMAAyqw8JQoABzZAwOFwDH4GAmTIAdXwezgaC5fHZlL2lBpewMyHAaHxGJ5A2sMFJLN5JhwyCCOEyAwxACs0H9ISzVEV2aY4AAZHlwM0Wq0gW1-dk83mZOAARTkEHg3stwpAXOseo1sx8cEyjNYtlgEpo+2QAA4RBiMxABhKWawNRnZWsaV6MQBHCPwQ1YqkgfoAWigcEcjkZ1jgjcoA8NOGNpqQ5pjGIGJkoAqFM6DIfDka9k59scCPmz1FzSGSGMFOEomSDHAgJhNGqg0DrIDkAwAKjgfFSp76afDzFBqAgOWAdQwAAgr+7IwAYIbRgMYhiEAA)
```sh
--parser babel
```

**Input:**
```jsx
test.each`
  id   | weaponChoice
  ${0} | ${'nunchucks'}     // why am I stripped?
  ${1} | ${'bo staff'}         why am I stripped?
  ${2} | ${'old blanket'}   /* why am I stripped? */
`('(test case $id) foo is bar when player attacks with $weaponChoice', () => {
  // test body
  expect(1).toBe(true);
});
```

**Output:**
```jsx
test.each`
  id   | weaponChoice
  ${0} | ${"nunchucks"}
  ${1} | ${"bo staff"}
  ${2} | ${"old blanket"}
`("(test case $id) foo is bar when player attacks with $weaponChoice", () => {
  // test body
  expect(1).toBe(true);
});

```

**Expected behavior:**
The text "why am I stripped?" should not be stripped.

I can see there is correct behaviour if I change `test.each` to `test.foo`.

**Comment by lydell** [comment]
@duncanawerbuck Just checking, are you 100% Jest allows such comments inside the template string?

**Comment by duncanawerbuck** [comment]
@lydell I don't see anything mentioned explicitly [in their docs](https://jestjs.io/docs/en/api#testeachtable-name-fn-timeout) about NOT including additional text in the template literal, and TBH I haven't had a look at the source code yet.

The tests that we write that **do** have those trailing comments ([playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuE8DOMB0cCGYAWABgDpQAEZAlgCYVkA+ZA7rgA7QDC+ElYcpFACTAADAF8GZYQHIoAVygE5YANZppEumQD025vgCeZHAFsyASTIYATpVas41APwCpwAIwTGMgEYQrMDgAZkEaWuFMhsZmljZ2Ds6uwgBMXm7SEAA2tD6ZOFAqcDBhFNoAVPpGphYBtvaOTmRl2qSEABTSbegwZGA4aHBSNACUZEEQ-pRoZD441voIZKx5BnDzODCBqtNMlDD4Uiw47FBcPHzSADRkbaMAvAB8ZMCuumTdMxDUBq5wAB4OMAwNruYaYGAQABCcC61jkcGGAG5SGIkSBLiAIKwYJRoGhkKA5tYIEwAApzBD4lA4ABuPGo6JAPmseEKMAAyqw8JQoABzZAwOFwDH4GAmTIAdXwezgaC5fHZlL2lBpewMyHAaHxGJ5A2sMFJLN5JhwyCCOEyAwxACs0H9ISzVEV2aY4AAZHlwM0Wq0gW1-dk83mZOAARTkEHg3stwpAXOseo1sx8cEyjNYtlgEpo+2QAA4RBiMxABhKWawNRnZWsaV6MQBHCPwQ1YqkgfoAWigcEcjkZ1jgjcoA8NOGNpqQ5pjGIGJkoAqFM6DIfDka9k59scCPmz1FzSGSGMFOEomSDHAgJhNGqg0DrIDkAwAKjgfFSp76afDzFBqAgOWAdQwAAgr+7IwAYIbRgMYhiEAA)) execute and pass/fail as they should.

My expectation is that stripping text in this way is not entirely in line with the Prettier philosophy of prioritising the prevention of loss of code.

I've had a look (not exhaustively!) for relevant discussion in other issues and I'm not sure if this particular scenario has been discussed or considered, but I understand this is an edge case though, so please don't take this as criticism. 😃 

If anyone feels that auto-stripping text in this way (where it *doesn't* strip it if you change `test.each...` to `test.anyRandomWord...`) is *desirable*, then I don't mind this issue being closed. Would be good to hear the reasoning though if possible.

I'd also like to add that Prettier is ❤️amazing❤️ and I'm very grateful to all contributors for the cognitive overhead it saves!!

**Comment by lydell** [comment]
I agree with you that losing code here is confusing. I would just never have thought of adding comments there so I wanted to make sure that they were actually valid before thinking of a solution. We should either support formatting them or at least not remove them, I think.

**Comment by duncanawerbuck** [comment]
Yes, making sure they're not lost is definitely a good thing.

In terms of HOW to format them, I think this could get tricky, since in theory, someone could potentially put anything there (of any length).

It's probably safest not to amend those strings at all ("those strings" being the content that was stripped in the above playground).
